container-core/src/main/java/com/yahoo/container/jdisc/state/MetricsPacketsHandler.java

```diff
@@ -0,0 +1,188 @@
+// Copyright 2018 Yahoo Holdings. Licensed under the terms of the Apache 2.0 license. See LICENSE in the project root.
+package com.yahoo.container.jdisc.state;
+
+import com.google.inject.Inject;
+import com.yahoo.collections.Tuple2;
+import com.yahoo.component.provider.ComponentRegistry;
+import com.yahoo.container.jdisc.state.StateHandler.JSONObjectWithLegibleException;
```

Comment:
Consider moving `JSONObjectWithLegibleException` to separate file

Reply:
done.